src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private ProductService productService;
+    private SortUtil sortUtil;
+    private DtoMapper<Product,
+            ProductResponseDto,
+            ProductRequestDto> mapper;
+
+    @GetMapping
+    @ApiOperation(value = "Get all products")
+    List<ProductResponseDto> findAll(
+            @RequestParam (defaultValue = "20")
+                @ApiParam(value = "number of items on page. default value is 20") Integer count,
+            @RequestParam (defaultValue = "0")
+                @ApiParam(value = "requested page") Integer page,
+            @RequestParam (defaultValue = "id")
+                @ApiParam(value = "sorting parameters given in form field:order") String sortBy) {
+        Sort sort = Sort.by(sortUtil.parse(sortBy));
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAll(pageRequest)
+                .stream()
+                .map(mapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/by-price")
+    @ApiOperation(value = "Get all products in within given price")
+    List<ProductResponseDto> findAllByPriceBetween(
+            @RequestParam (defaultValue = "20")
+                @ApiParam(value = "number of items on page. default value is 20") Integer count,
+            @RequestParam (defaultValue = "0")
+                @ApiParam(value = "requested page") Integer page,
+            @RequestParam (defaultValue = "id")
+                @ApiParam(value = "sorting parameters given in form field:order") String sortBy,
+            @RequestParam
+                @ApiParam(value = "lower border of given price") BigDecimal from,
+            @RequestParam
+                @ApiParam(value = "upper border of given price") BigDecimal to) {
+        Sort sort = Sort.by(sortUtil.parse(sortBy));
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAllByPriceBetween(from, to, pageRequest)
+                .stream()
+                .map(mapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    ProductResponseDto findById(@PathVariable
+                                    @ApiParam(value = "Id of item") Long id) {
+        return mapper.toDto(productService.find(id));
+    }
+
+    @PostMapping
+    @ApiOperation(value = "Create new product")
+    ProductResponseDto save(@RequestBody ProductRequestDto dto) {

```

Comment:
add description for request bodies, for all controller methods that have it
https://stackoverflow.com/questions/64645528/java-spring-boot-openapi-3-how-to-add-description-for-requestbody